### ISSUE TITLE
False positive E0277 on Self type with Sized bound 

### ISSUE THREAD

**Issue by Exidex** [intellij-rust/intellij-rust#4941]
## Environment

* **IntelliJ Rust plugin version:** 0.2.115.2157-193
* **Rust toolchain version:** 1.41.0
* **CLion:** 2019.3.3
Build #CL-193.6015.37, built on January 20, 2020
Runtime version: 11.0.5+10-b520.30 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.3.18-1-MANJARO

## Problem description

False positive "the trait bound `Self: std::marker::Sized` is not satisfied [E0277] `Self` does not have a constant size known at compile-time" on Self parameter type or return type, that has 'Sized' bound 

## Steps to reproduce
<details><summary>Code as plain text</summary>
<p>

```rust
trait TheTrait {

    fn fine(self) -> Self where Self: Sized;
    fn errors(self) -> Self where Self: Sized {
        self
    }
    fn also_errors(self, other: Self) -> Self where Self: Sized {
        self
    }
}
```
</p>
</details>

![Screenshot from 2020-02-08 00-41-05](https://user-images.githubusercontent.com/16986685/74073802-ed9ff400-4a0b-11ea-9b8d-a0c21292ca03.png)

**Comment by Undin** [comment]
Duplicate of #2530
